### PR TITLE
Portable backtrace mechanism

### PR DESCRIPTION
**Pull request by cmnrd**
This makes the backtrace mechanism that we currently use for assertions portable. The current mechanism was implemented under the assumption, that all Linux distributions provide `execinfo.h`, which is not the case (e.g. Alpine has dropped libexecinfo). This PR provides a more robust implementation that uses [FindBacktrace](https://cmake.org/cmake/help/latest/module/FindBacktrace.html) to find the correct backtrace implementation.

This change allows us to simplify the docker support in LF.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -30,7 +30,10 @@ if (NOT DEFINED LF_REACTOR_CPP_SUFFIX)
   endif()
 endif()
 
-find_package (Threads)
+find_package(Threads)
+
+find_package(Backtrace)
```
Comment (tanneberger) [inline CMakeLists.txt:35]: Why not hinding it behind some build flag?

File: include/reactor-cpp/assert.hh [hunk 2]
```diff
@@ -37,6 +32,33 @@ constexpr bool runtime_assertion = true;
...
+
+namespace reactor {
+
+constexpr std::size_t MAX_TRACE_SIZE{16};
+
+inline void print_backtrace() {
+  std::array<void*, MAX_TRACE_SIZE> trace{nullptr};
+  int size = backtrace(trace.data(), MAX_TRACE_SIZE);
+  char** messages = backtrace_symbols(trace.data(), size);
+  for (int i{0}; i < size; i++) {
```
Comment (tanneberger) [inline include/reactor-cpp/assert.hh:50]: I know its valid, but this just doesn't look right :D

Comment (cmnrd) [reply include/reactor-cpp/assert.hh:50]: What does not look right?

File: CMakeLists.txt (hunk 1 above)
Comment (cmnrd) [reply CMakeLists.txt:35]: I assume you mean 'hiding', but I don't understand what you would like to hide? This PR effectively enables backtracing support automatically if an implementation of `backtrace()` is found.

File: include/reactor-cpp/assert.hh (hunk 2 above)
Comment (tanneberger) [reply include/reactor-cpp/assert.hh:50]:
```cpp
int i{0}; 
```

**Pull request merged**